### ISSUE TITLE
Improve the configure script's error messages on missing OpenSSL

### ISSUE THREAD

**Issue by solardiz** [openwall/john#4061]
As prompted by #4058:

Can we somehow improve the `configure` error message "JtR requires libssl being installed" to be more helpful?

Maybe we should refer to doc/INSTALL from configure's error message, and to INSTALL-UBUNTU and INSTALL-FEDORA and INSTALL-WINDOWS from there. Right now, doc/INSTALL does mention needing libssl and gives Debian/Ubuntu package names as an example, but it does not refer to the other INSTALL-something files - perhaps we should improve that.

Maybe we should rename INSTALL-FEDORA to INSTALL-REDHAT, so that people would know it also applies to RHEL and CentOS? OTOH, now there's also Red Hat CoreOS, and I don't know what instructions would apply to it. And people might not associate their CentOS with Red Hat. And there's the yum vs. dnf difference for RHEL up to 7 vs. newer Fedora, although I think there's still a yum wrapper even in the latest?

Separately, we should also work towards removing the dependency on OpenSSL.

**Comment by solardiz** [comment]
> doc/INSTALL [...] does not refer to the other INSTALL-something files - perhaps we should improve that.

I've just added those references to there.

> Separately, we should also work towards removing the dependency on OpenSSL.

This dependency is now optional and documented as such in `INSTALL`.

However, we should still improve the `configure` script's error messages on missing OpenSSL - these should point to more information on installing OpenSSL and on disabling its usage. I leave this issue open for that.

**Comment by magnumripper** [comment]
There's also this
```
libgmp (PRINCE mode and faster SRP formats)  yes
128-bit integer (faster PRINCE mode) ....... yes
libz (pkzip and some other formats) ........ yes
libbz2 (gpg2john extra decompression logic)  yes
libpcap (vncpcap2john and SIPdump) ......... yes
```
We could want to add a line something like
```
OpenSSL (many formats) ..................... yes
```

**Comment by solardiz** [comment]
> We could want to add a line something like
> 
> ```
> OpenSSL (many formats) ..................... yes
> ```

Yes, we need to add something like this. If we continue to also support CommonCrypto, then perhaps it should be the library name in the final field.

magnum, I am leaving this issue for you to work on. ;-)
